context.go

```diff
@@ -58,7 +58,7 @@ func NewContext(handle *Handle) *Context {
 // timeout value. It returns the matches as a JSON string (usually opaquely used) along with the corresponding
 // actions in any. In case of an error, matches and actions can still be returned, for instance in the case of a
 // timeout error. Errors can be tested against the RunError type.
-func (context *Context) Run(addressesToData map[string]any, timeout time.Duration) (matches []byte, actions []string, err error) {
+func (context *Context) Run(addressesToData map[string]any, timeout time.Duration) (res Result, err error) {
```

Comment:
The whole function does not need named return so we can remove them.

Reply:
```suggestion
func (context *Context) Run(addressesToData map[string]any, timeout time.Duration) (Result, error) {
```

Reply:
Named return type here allows default initialisation for early return cases of `res`. Another way to do it would be simply `var res Result` at the top of the func, which is less idiomatic.